src/views/Scenario/ScenarioHook.js

```diff
@@ -5,26 +5,50 @@ import { useTranslation } from 'react-i18next';
 import { useSetApplicationErrorMessage } from '../../state/app/hooks';
 import { useDatasets } from '../../state/datasets/hooks';
 import { useOrganizationId } from '../../state/organizations/hooks';
-import { useCurrentSimulationRunnerData, useSetSimulationRunnerValidationStatus } from '../../state/runner/hooks';
+import {
+  useCurrentSimulationRunnerData,
+  useSetSimulationRunnerValidationStatus,
+  useCurrentSimulationRunnerDatasetBases,
+} from '../../state/runner/hooks';
 import { useWorkspaceId } from '../../state/workspaces/hooks';
 
 export const useScenario = () => {
   const { t } = useTranslation();
+  const setApplicationErrorMessage = useSetApplicationErrorMessage();
+
   const currentScenarioData = useCurrentSimulationRunnerData();
   const organizationId = useOrganizationId();
   const workspaceId = useWorkspaceId();
   const datasets = useDatasets();
-  const currentScenarioDatasetName = useMemo(() => {
-    const scenarioDatasets = currentScenarioData?.datasets?.bases ?? [];
-    if (scenarioDatasets.length === 0) return t('views.scenario.text.nodataset', 'None');
+  const currentScenarioDatasetsBases = useCurrentSimulationRunnerDatasetBases();
 
-    return (
-      datasets?.find((dataset) => dataset.id === scenarioDatasets?.[0])?.name ??
-      t('views.scenario.text.datasetNotFound', 'Not found')
+  const { datasetBases, datasetCannotBeAccessed } = useMemo(() => {
+    const datasetBases = datasets.filter((dataset) => currentScenarioDatasetsBases?.includes(dataset.id));
+    const datasetCannotBeAccessed = currentScenarioDatasetsBases.filter(
+      (datasetId) => !datasetBases.some((dataset) => dataset.id === datasetId)
     );
-  }, [currentScenarioData?.datasets?.bases, datasets, t]);
+
+    if (currentScenarioDatasetsBases.length > 0 && datasetBases.length < currentScenarioDatasetsBases.length) {
+      const errorMessage = t(
+        'commoncomponents.banner.cannotAccessDatasetBases',
+        'For the scenario "{{scenarioName}} ({{scenarioId}})", you don\'t have access to its bases dataset ' +
+          '"{{datasetBasesId}}". Either the dataset no longer exists or you don\'t have sufficient permissions.',
+        {
+          datasetBasesId: currentScenarioDatasetsBases[0],
```

Comment:
If there are multiple base datasets used by the scenario, the missing dataset is not necessarily the first one in the list (`currentScenarioDatasetsBases[0]`). Can't we check if `datasetCannotBeAccessed` is a non-empty array instead?